libraries/AP_Common/Bitmask.h

```diff
@@ -50,22 +57,42 @@ class Bitmask {
     }
 
     // return true if given bitnumber is set
-    bool get(uint16_t bit) {
+    bool get(uint16_t bit) const {
         uint16_t word = bit/32;
         uint8_t ofs = bit & 0x1f;
         return (bits[word] & (1U << ofs)) != 0;
     }
 
     // return true if all bits are clear
-    bool empty(void) {
+    bool empty(void) const {
         for (uint16_t i=0; i<numwords; i++) {
             if (bits[i] != 0) {
                 return false;
             }
         }
         return true;
     }
-    
+
+    // return number of bits set
+    uint16_t count() const {
+        uint16_t sum = 0;
+        for (uint16_t i=0; i<numwords; i++) {
+            if (sizeof(bits[i]) <= sizeof(int)) {
+                sum += __builtin_popcount(bits[i]);
+            } else if (sizeof(bits[i]) <= sizeof(long)) {
+                sum += __builtin_popcountl(bits[i]);
+            } else {
+                sum += __builtin_popcountll(bits[i]);
+            }
+        }
+        return sum;
+    }
+
+    // return number of bits available (may be slightly larger than number requested in constructor)
```

Comment:
I'd rather this was correct. Possible uses for bitmasks include because they indirectly lookup information about something else, and iterating over every bit of a bitmask is a very natural action, and going to size makes sense. But with any indirection (IE the bitmask is used to lookup what page of memory is free) this all of a sudden could be walking off the end. If this is needed, then actually storing the count in a uint32_t makes more sense.